automation/patch_igz/patch_remote.py

```diff
@@ -40,101 +40,120 @@ class Consts:
+                        target_to_built_images["log-collector"],
+                    )
 
-            self._wait_deployment_ready()
+                # Reset or rollout deployment as necessary
+                if self._reset_db:
+                    self._reset_mlrun_db()
+                else:
+                    self._rollout_deployment()
 
-        finally:
-            out = self._exec_remote(
-                [
-                    "kubectl",
-                    "-n",
-                    "default-tenant",
-                    "get",
-                    "pods",
-                ],
-            )
+                self._wait_deployment_ready()
 
-            for line in out.splitlines():
-                if self.Consts.api_container in line:
-                    logger.info(line)
-                elif self.Consts.log_collector_container in line:
-                    logger.info(line)
+            finally:
+                # Check status of pods after deployment
+                out = self._exec_remote(
+                    ["kubectl", "-n", "default-tenant", "get", "pods"]
+                )
+                for line in out.splitlines():
+                    if (
+                        self.Consts.api_container in line
+                        or self.Consts.log_collector_container in line
+                    ):
+                        logger.info(line)
 
-            self._disconnect_from_node()
+                self._disconnect_from_node()
 
         logger.info(
-            "Deployed branch successfully! Yay! (Note this may not survive system restarts)"
+            "Deployed branch successfully! (Note: This may not survive system restarts)"
         )
 
     def _docker_login_if_configured(self):
         registry_username = self._config.get("REGISTRY_USERNAME")
         registry_password = self._config.get("REGISTRY_PASSWORD")
-        registry_url = self._config.get("REGISTRY_URL")
-        if not registry_username:
-            return
-        command = [
-            "docker",
-            "login",
-            registry_url or "",
-            "--username",
-            registry_username,
-            "--password-stdin",
-        ]
-        completed_process = subprocess.run(
-            command, input=registry_password.encode() + b"\n", capture_output=True
-        )
-        if completed_process.returncode != 0:
-            raise RuntimeError(
-                f"Failed to login to docker registry. Error: {completed_process.stderr}"
+        docker_registry = self._config.get("DOCKER_REGISTRY")
+        if registry_username is not None:
+            self._exec_local(
+                [
+                    "docker",
+                    "login",
+                    docker_registry or "",
+                    "--username",
+                    registry_username,
+                    "--password",
+                    registry_password,
+                ],
+                live=True,
```

Comment:
suggest to undo to my changes so allow reading password from stdin and not as plain text